README.md

```diff
@@ -2,68 +2,117 @@
-2. For Linux users only, [Docker Compose](https://docs.docker.com/compose/install/)
+2. [Docker Compose](https://docs.docker.com/compose/install/) (Must be installed seperately on Linux)*
 3. [Ruby](https://www.ruby-lang.org/en/documentation/installation/)
+4. [Yarn](https://yarnpkg.com/lang/en/docs/install/)
+
+* If you are running OSx or Windows, your Docker installation will have 
+come with docker-compose. 
+
+### Clone the repo and run the init script 
+
+We have an init script that installs a linter (rubocop) and sets up a 
+precommit hook making the linter run automatically whenever you commit.
+It is kept under our `script` directory.
+
+After cloning the repo and navigating into the cloned directory, run the init script
+in your shell. To do so in bash, for example, invoke:
 
-### Clone repo and run init script 
-We have an init script that installs a linter (rubocop), as well as sets up a 
-precommit hook to have the linter automatically run whenever you make a commit.
-Clone this repo, navigate into the cloned directory, and run 
 ```
-./script/init-setup.sh
+bash script/init-setup.sh
 ```
 
-### Running in a Docker Container
-To launch the app, simply navigate into the cloned directory and run
+### Running the backend with Docker
+
+To begin, make sure that Docker is running on your system. Once it is, invoke
+
 ```
 docker-compose up
 ```
 
-This will launch two containers: a rails app, and a postgres database. Access
-the rails app by navigating to `http://localhost:3000`
+from the cloned directory.
+
+This will launch two containers: a rails app, and a postgres database. You can 
+access the rails app from your browser by navigating to `http://localhost:3000`. 
+(Bare in mind that in tapp, rails is meant only to serve the API, not views.)
+
+If you already have an existing image running on your machine that you would
+like to rebuild, run
 
-If you have an existing image on your machine that needs to be updated, run
 ```
 docker-compose up --build
 ```
 
-To view the STDOUT from a docker container of a running server, you can do
-`docker-compose logs -tf <image>`, like: `docker-compose logs -tf tapp`.
+To view the STDOUT from a docker container of a running server, you can invoke
+
+```docker-compose logs -tf <image>```
+
+### Navigating into the containers from the command line
+
+Currently, we define two services under docker-compose: tapp and db. If you would
+like to interact with either container from the command line, you can do so by 
+invoking:
 
-To access the dockerized app, simply run
 ```
-docker-compose exec tapp sh
+docker-compose exec [service] sh
 ```
 
-### Initializing DB
-To create and setup your local development database, simply navigate into the
-rails container and run the following rake task. It will create your local database,
-run all migrations, and populate the db with seed data. 
+To interact with the rails app, invoke `docker-compose exec tapp sh` and to
```

Comment:
what means "interact" ??

Reply:
It means that you can enter an interactive terminal session within the container, I took the wording from docker resources. Under such a session, you could run our rails specs, for example. Do you want this to be clarified further in the document ?